mlflow/types/utils.py

```diff
@@ -345,13 +345,13 @@ def _validate_input_dictionary_contains_only_strings_and_lists_of_strings(data)
     for key, value in data.items():
         if not value_type:
             value_type = type(value)
-        if isinstance(key, bool):
+        if isinstance(key, bool) or not isinstance(key, (str, int)):
             invalid_keys.append(key)
-        elif not isinstance(key, (str, int)):
-            invalid_keys.append(key)
-        if isinstance(value, list) and not all(isinstance(item, (str, bytes)) for item in value):
```

Comment:
`if` here should be `elif`. It's weird to append the same key twice to `invalid_keys`.

Reply:
Yes, I was confused with this but thought it was intentional. Let me fix it.